supervision/detection/core.py

```diff
@@ -504,6 +534,60 @@ class names. If provided, the resulting Detections object will contain
                 class_id=class_ids,
                 data=data,
             )
+        elif "segments_info" in transformers_results:
```

Comment:
Personally, I would move `segments_info = transformers_results["segments_info"]` extraction to ` if "segmentation" in transformers_results:` and `elif "png_string" in transformers_results:` blocks. We will get one indention less.